File: 프로그래머스/0/181935. 홀짝에 따라 다른 값 반환하기/홀짝에 따라 다른 값 반환하기.java
import java.util.stream.*;

class Solution {
    public int solution(int n) {
        return IntStream.rangeClosed(0, n)
            .filter(e -> n%2 == 0 ? e%2 == 0 : e%2 != 0)
            .map(e -> n%2 == 0 ? e*e : e)
            .sum();
    }
}

File: 프로그래머스/0/181935. 홀짝에 따라 다른 값 반환하기/홀짝에 따라 다른 값 반환하기.py
def solution(n):
        return intSum(n+1) if n % 2 == 0 else oddSum(n+1)

def oddSum(n):
    return sum(range(1, n, 2))
 
def intSum(n):
    return sum([x * x for x in range(0, n, 2)])